packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -747,15 +794,12 @@ export default function makeKernelKeeper(storage, kernelSlog) {
   }
 
   function getAllVatIDs() {
-    const nextID = Nat(Number(getRequired(`vat.nextID`)));
-    const vatIDs = [];
-    for (let i = FIRST_VAT_ID; i < nextID; i += 1) {
-      const vatID = makeVatID(i);
-      if (storage.has(`${vatID}.o.nextID`)) {
-        vatIDs.push(vatID);
-      }
-    }
-    return harden(vatIDs);
+    return harden(
+      vatC
+        .getAllIDs()
+        .map(makeVatID)
+        .filter(vatID => storage.has(`${vatID}.o.nextID`)),
+    );
```

Comment:
This needs to go back to something more like its original form as it does not scale to the gajillion vats case -- the main thing is to get rid of `getAllIDs()`.

Reply:
Ah, I wasn't thinking of that. Reverted!